src/file_drop/mod.rs

```diff
@@ -0,0 +1,147 @@
+}
+impl FileDropListener {
+    pub(crate) fn new(handlers: (Option<FileDropHandler>, Option<FileDropHandler>)) -> FileDropListener {
+        debug_assert!(handlers.0.is_some() || handlers.1.is_some(), "Tried to create a FileDropListener with no file drop handlers!");
+        FileDropListener {
+            handlers,
+            active_file_drop: Cell::new(None),
+        }
+    }
+
+    // Called when a file drop event occurs. Bubbles the event up to the handler.
+    // Return true to prevent the OS' default action for the file drop.
+    pub(crate) fn file_drop(&self, event: FileDropEvent, paths: Option<Vec<PathBuf>>) -> bool {
+        let paths = match event {
+
+            FileDropEvent::Hovered => {
+                if paths.is_none() || paths.as_ref().unwrap().is_empty() {
+                    debug_assert!(false, "FileDropEvent::Hovered received with missing or empty paths list!");
+                    return false;
+                }
+                paths.unwrap()
+            },
+
+            _ => match paths {
+
+                Some(paths) => paths,
+                
+                None => match self.active_file_drop.take() {
+                    None => {
+                        debug_assert!(false, "Failed to retrieve paths list from memory for this file drop event!");
+                        return false;
+                    },
+                    Some(status) => match status {
+                        FileDropStatus::Hovered(paths) => paths,
+                        FileDropStatus::Dropped(paths) => paths,
+                        FileDropStatus::Cancelled(paths) => paths
+                    }
+                }
+
+            }
+
+        };
+
+        let new_status = match event {
+            FileDropEvent::Hovered => FileDropStatus::Hovered(paths),
+            FileDropEvent::Dropped => FileDropStatus::Dropped(paths),
+            FileDropEvent::Cancelled => FileDropStatus::Cancelled(paths)
+        };
+
+        self.active_file_drop.set(Some(new_status.clone()));
+        self.call(new_status)
+    }
+
+    fn call(&self, status: FileDropStatus) -> bool {
+        // Kind of silly, but the most memory efficient
+        let mut prevent_default = false;
+        match self.handlers.0 {
+            Some(ref webview_file_drop_handler) => {
+                match self.handlers.1 {
+                    Some(ref app_file_drop_handler) => {
+                        prevent_default = webview_file_drop_handler.call(status.clone()) | app_file_drop_handler.call(status);
+                    },
+                    None => prevent_default = webview_file_drop_handler.call(status)
+                }
+            },
+            None => {
+                match self.handlers.1 {
+                    Some(ref app_file_drop_handler) => prevent_default = app_file_drop_handler.call(status),
+                    None => {}
+                }
+            }
+        }
+        prevent_default
+    }
+}
```

Comment:
missing EOF empty line

Reply:
Shall we add an .editorconfig file to enforce this?

Reply:
```json
root = true

[*]
indent_style = space
indent_size = 4
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```